examples/basic_auth.go

```diff
@@ -8,7 +8,12 @@ import (
 
 // This example shows how to create a client with username and password.
 func basicAuthExample() {
-	git, err := gitlab.NewBasicAuthClient(nil, "https://gitlab.company.com", "svanharmelen", "password")
```

Comment:
Why not turn everything into options and have a single constructor function? :) Passing `nil` is not that nice and is begging to be an optional parameter. Same with auth - pick an option that works for you.

Reply:
I agree that putting the optional HTTP client into an option indeed makes sense, so I'll adjust that one. But I'll keep the dedicated new client functions to make it more clear what type of clients you can create.